src/overlays/actors/ovl_En_Diving_Game/z_en_diving_game.c

```diff
@@ -39,57 +67,522 @@ static ColliderCylinderInit D_809EF0B4 = {
+static u64* sEyeTextures[] = {
+    0x06003E40,
+    0x06004640,
+    0x06004E40,
+};
+
+extern FlexSkeletonHeader D_0600BFA8;
+extern AnimationHeader D_06002FE8;
+extern AnimationHeader D_0600219C;
+
+void EnDivingGame_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDivingGame* this = THIS;
+
+    this->actor.gravity = -3.0f;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 30.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600BFA8, &D_06002FE8, this->jointTable, this->morphTable, 20);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 素もぐりＧＯ ☆☆☆☆☆ \n" VT_RST);
+    this->actor.room = -1;
+    this->actor.scale.x = 0.01f;
+    this->actor.scale.y = 0.012999999f;
+    this->actor.scale.z = 0.0139999995f;
+    if (D_809EF0B0 != 0) {
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ もういてる原 ☆☆☆☆☆ \n" VT_RST);
+        this->unk_31F = 1;
+        Actor_Kill(&this->actor);
+    } else {
+        D_809EF0B0 = 1;
+        this->actor.targetMode = 0;
+        this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+        this->actionFunc = func_809EDCB0;
+    }
+}
+
+void EnDivingGame_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDivingGame* this = THIS;
+
+    if (this->unk_31F == 0) {
+        gSaveContext.timer1State = 0;
+    }
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnDivingGame_SpawnRuppy(EnDivingGame* this, GlobalContext* globalCtx) {
+    EnExRuppy* attached;
+    Vec3f rupeePos;
+
+    rupeePos.x = (Rand_ZeroOne() - 0.5f) * 30.0f + this->actor.world.pos.x;
+    rupeePos.y = (Rand_ZeroOne() - 0.5f) * 20.0f + (this->actor.world.pos.y + 30.0f);
+    rupeePos.z = (Rand_ZeroOne() - 0.5f) * 20.0f + this->actor.world.pos.z;
+    attached = (EnExRuppy*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_EX_RUPPY,
+                                              rupeePos.x, rupeePos.y, rupeePos.z, 0,
+                                              (s16)Rand_CenteredFloat(3500.0f) - 1000, this->unk_2A6, 0);
+    if (attached != NULL) {
+        attached->actor.speedXZ = 12.0f;
+        attached->actor.velocity.y = 6.0f;
+    }
+}
+
+s32 func_809EDB08(EnDivingGame* this, GlobalContext* globalCtx) {
+    s32 var;
+
+    if (gSaveContext.timer1State == 10 && Gameplay_InCsMode(globalCtx) == 0) {
+        gSaveContext.timer1State = 0;
+        func_800F5B58();
+        func_80078884(NA_SE_SY_FOUND);
+        this->actor.textId = 0x71AD;
+        func_8010B680(globalCtx, this->actor.textId, NULL);
+        this->unk_292 = 5;
+        this->unk_31E = this->unk_2A8 = this->unk_29C = this->unk_2A2 = this->unk_2A4 = 0;
+        func_8002DF54(globalCtx, NULL, 8);
+        this->actionFunc = func_809EE048;
+        return 1;
+    } else {
+        var = 5;
```

Comment:
```suggestion
        s32 var = 5;
        
```
declaring down here makes more sense I think.